lib/redis/time_series/sample.rb

```diff
@@ -3,37 +3,28 @@ class Redis
   class TimeSeries
     # A sample is an immutable value object that represents a single data point within a time series.
     class Sample
-      using TimeMsec
 
       # @return [Time] the sample's timestamp
-      attr_reader :time
+      attr_reader :time,:ts_msec
       # @return [BigDecimal] the decimal value of the sample
-      attr_reader :value
+      attr_accessor :value
 
       # Samples are returned by time series query methods, there's no need to create one yourself.
       # @api private
       # @see TimeSeries#get
       # @see TimeSeries#range
       def initialize(timestamp, value)
-        @time = Time.from_msec(timestamp)
+        @ts_msec = timestamp
+        @time = Time.at(timestamp / 1000)
```

Comment:
Debating whether it's worth extracting a constant for `1000` here... not that the number of seconds in a millisecond will change, but might be good to keep track of the various locations it's used.